text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,772 @@
+|| ┌───────────────────┐ |         | ┌──────────────────┐  ||                                 
+|| │ CFN stack refactor| |         | │  Retain Policy   |  ||
+|| |    validation     | |         | |    validation    |  ||                                   
+|| └───────────────────┘ |         | └──────────────────┘  ||
+||           │           |         |            │          ||    
+||           ▼           |         |            ▼          ||    
+|| ┌───────────────────┐ |         |┌─────────────────────┐||                                 
+|| │ Execute Stack     | |         |│     Execute         |||  
+|| |    refactoring    | |         || Retain-Remove-Import||| 
+|| └───────────────────┘ |         |└─────────────────────┘|| 
+|└───────────────────────┘         └───────────────────────┘|
+|           │                               │               | 
+|           └───────────────────────────────┘               |
+|                          |                                |
+|                          ▼                                |
+|               ┌─────────────────────┐                     |
+|               │        Done         │                     |
+|               │                     │                     |
+|               └─────────────────────┘                     |
+└───────────────────────────────────────────────────────────┘
+```
+
+### Examples
+
+Suppose your CDK application has a single stack, called `MyStack`, containing a
+DynamoDB `Table` construct with replica. Now you want to migrate to use existing
+stack to use the latest construct `TableV2` which offers more advanced features
+to define table replica configurations.
+
+```ts
+const table = new dynamodb.Table(this, 'MyTable', {
+  partitionKey: {
+    name: 'PK',
+    type: AttributeType.STRING,
+  },
+  tableName: 'MyTable',
+  replicationRegions: ['us-west-2']
+});
+```
+
+CDK provides a DynamoDB table migration guide that guides users on the code migration
+processes. Assuming users follow the migration guide correctly and update their CDK code
+to use `TableV2` construct.
+
+In case of DynamoDB Table migration, the new CDK code would become:
+
+```ts
+const table = new TableV2(this, 'MyTable', {
+  partitionKey: {
+    name: 'PK',
+    type: AttributeType.STRING,
+  },
+  tableName: 'MyTable',
+  deletionProtection: true,
+  replicas: [
+    {
+      region: 'us-west-2',
+    }
+  ]
+});
+```
+
+> Note that directly deploying the above CDK would cause CloudFormation
+> to delete your existing table and create a new one, due to differences in logical IDs
+> and resource types.
+
+The next step is to run the CDK CLI command `cdk construct-upgrade`. This command
+will trigger the validation process to make sure the changes are valid and safe to deploy.
+This migration will use the `Retain-Remove-Import` migration strategy because the underlying
+resource type has changed from using a custom resource to native CloudFormation resource.
```

Comment:
I think you mean from using `AWS::DynamoDB::Table` to using `AWS::DynamoDB::GlobalTable`?